### PR TITLE
Remove ManagedVector from filters

### PR DESCRIPTION
**Pull request by WeiqunZhang**

### CODE REVIEW

**Review by ax3l — APPROVED** [review]
Thank you, had to look twice with the `old_s` manipulation (local var, all good).

Update: uh wait, actually that might cause a lifetime issue.

File: Source/Filter/BilinearFilter.cpp [hunk 1]
```diff
@@ -46,8 +46,9 @@ namespace {
         }
         // we use old_s here to make sure the stencil
         // is corrent even when npass = 0
-        stencil = old_s;
-        stencil[0] *= 0.5; // because we will use it twice
+        old_s[0] *= 0.5; // because we will use it twice
+        stencil.resize(old_s.size());
+        Gpu::copyAsync(Gpu::hostToDevice,old_s.begin(),old_s.end(),stencil.begin());
```
Comment (ax3l) [inline Source/Filter/BilinearFilter.cpp:51]: I fear that `old_s` will run out of scope and we could be touching invalid source memory here during the copy.

Comment (WeiqunZhang) [reply Source/Filter/BilinearFilter.cpp:51]: Yes, I believe you are right.

**Review by ax3l — APPROVED** [review]
Thanks for the quick update.

**Review by RemiLehe — APPROVED** [review]

**Pull request merged**